File: lamp-blog-web/src/main/java/space/lingu/lamp/web/domain/staff/dto/StaffInfo.java
/*
 * Copyright (C) 2023 RollW
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package space.lingu.lamp.web.domain.staff.dto;

import space.lingu.lamp.web.domain.staff.Staff;
import space.lingu.lamp.web.domain.staff.StaffType;
import space.lingu.lamp.web.domain.user.UserIdentity;

import java.util.Set;

/**
 * @author RollW
 */
public record StaffInfo(
        long userId,
        UserIdentity userIdentity,
        String employeeId,
        Set<StaffType> types,
        long createTime,
        long updateTime,
        boolean allowUser,
        boolean deleted
) {
    public static StaffInfo from(Staff staff, UserIdentity userIdentity) {
        if (staff == null) {
            return null;
        }
        return new StaffInfo(
                staff.getUserId(),
                userIdentity,
                staff.getEmployeeId(),
                staff.getTypes(),
                staff.getCreateTime(),
                staff.getUpdateTime(),
                staff.isAllowUser(),
                staff.isDeleted()
        );
    }

}

File: lamp-blog-web/src/main/java/space/lingu/lamp/web/domain/staff/StaffInfo.kt
/*
 * Copyright (C) 2023 RollW
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package space.lingu.lamp.web.domain.staff

import space.lingu.lamp.user.UserIdentity

/**
 * @author RollW
 */
data class StaffInfo(
    val id: Long,
    val userIdentity: UserIdentity,
    override val types: Set<StaffType>,
    private val createTime: Long,
    private val updateTime: Long,
    val allowUser: Boolean,
    val deleted: Boolean
) : AttributedStaff {
    override val staffId: Long
        get() = id

    override val userId: Long
        get() = userIdentity.userId

    override fun getCreateTime(): Long = createTime

    override fun getUpdateTime(): Long = updateTime

    companion object {
        @JvmStatic
        fun from(staff: Staff, userIdentity: UserIdentity): StaffInfo {
            return StaffInfo(
                staff.id,
                userIdentity,
                staff.types,
                staff.createTime,
                staff.updateTime,
                staff.isAllowUser,
                staff.isDeleted
            )
        }

        @JvmStatic
        fun Staff.toStaffInfo(userIdentity: UserIdentity): StaffInfo {
            return from(this, userIdentity)
        }
    }
}